lib/content-services/src/lib/dialogs/node-lock.dialog.ts

```diff
@@ -35,7 +33,7 @@ export class NodeLockDialogComponent implements OnInit {
     node: Node = null;
     nodeName: string;
 
-    _nodesApi: NodesApi;
+    private _nodesApi: NodesApi;
     get nodesApi(): NodesApi {
         this._nodesApi = this._nodesApi ?? new NodesApi(this.alfrescoApi.getInstance());
```

Comment:
Just curious, but wouldn't it be better to just have 
```
private _nodesApi;;
get nodesApi(): NodesApi {
    return this._nodesApi;
}
constructor (alfrescoApi: AlfrescoApiService, ...) {
   this._nodesApi = new NodesApi(this.alfrescoApi.getInstance());
}
```
This way we can avoid the check for the existence of _nodesApi and the needless assignment of `this._nodesApi = this._nodesApi`  everytime the getter is called?

Since this seems like the only usage of the alfrescoApi object in this class, we would also not needlessly have to keep the object of AlfrescoApiService in memory  while this component is active (see lack of 'private' keyword in constructor argument).

Reply:
this is a deferred/lazy init, the reason is when Angular builds all singletons for services, JS-API is not yet initialised

Reply:
so we build the NodesApi instance on demand, when UI component is rendered, and that means auth and init was done before

Reply:
Ah, that makes sense. Thanks!!